pkg/proxyserver/controller/controller.go

```diff
@@ -216,6 +216,19 @@ func (c *ProxyServiceInfoController) generateServiceInfo(cm *corev1.ConfigMap) (
 		return nil, fmt.Errorf("failed to get secret in configmap %s/%s, %v", cm.Namespace, cm.Name, err)
 	}
 
+	caConfigMapNamespace, caConfigMapName, err := cache.SplitMetaNamespaceKey(cm.Data["caConfigMap"])
+
+	if err != nil {
+		return nil, fmt.Errorf("the caConfigMap format is wrong in configmap %s/%s, %v", cm.Namespace, cm.Name, err)
+	}
+	if caConfigMapNamespace == "" {
+		caConfigMapNamespace = serviceNamespace
+	}
+	caConfigmap, err := c.client.CoreV1().ConfigMaps(caConfigMapNamespace).Get(context.TODO(), caConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get configmap in configmap %s/%s, %v", cm.Namespace, cm.Name, err)
+	}
+
```

Comment:
Maybe we should also validate that the configmap has the `service-ca.crt` data field.  We can show a better error message if it doesn't have it.